### PR TITLE
Jetty 9.4.x 5859 classloader leak queuedthreadpool

### PR DESCRIPTION
**Pull request by janbartel**
See #5859

Ensure that QueuedThreadPool threads have their initial context classloader explicitly set, rather than defaulting to whatever the calling thread is using.

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java [hunk 1]
```diff
@@ -689,6 +689,7 @@ public Thread newThread(Runnable runnable)
         thread.setDaemon(isDaemon());
         thread.setPriority(getThreadsPriority());
         thread.setName(_name + "-" + thread.getId());
+        thread.setContextClassLoader(QueuedThreadPool.class.getClassLoader());
```
Comment (sbordet) [inline jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java]: Do we really want the `QueuedThreadPool` classloader? What if it's a subclass loaded by a different classloader?

File: jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java [hunk 2]
```diff
@@ -833,6 +834,24 @@ public void testDump() throws Exception
...
+        tp.setThreadsPriority(Thread.NORM_PRIORITY - 1);
+        try (StacklessLogging stackless = new StacklessLogging(QueuedThreadPool.class))
+        {
+            tp.start();
+            tp.execute(() ->
+            {
+                assertThat(Thread.currentThread().getContextClassLoader(), Matchers.equalTo(QueuedThreadPool.class.getClassLoader()));
+            });
+        }
+    }
```
Comment (sbordet) [inline jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java:859]: This test would have passed also before the change, right?
The real test should really create a different classloader, use a pooled thread, change its classloader, spawn _another_ thread and verify the spawned thread does not inherit the context classloader.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java (hunk 2 above)
Comment (janbartel) [reply jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java:859]: Made the test even simpler.

**Review by gregw — APPROVED** [review]
LGTM

**Review by sbordet — APPROVED** [review]

**Comment by janbartel** [comment]
@gregw / @sbordet  reviewing the OP's info from issue #5859, it seems to me that the QueuedThreadPool threads are also vulnerable to the problem he reported in the same issue regarding the ShutdownThread: namely the inheritance by a new Thread() of the calling AccessContext, containing a reference to the ProtectionDomain which has a reference to the current contextClassloader.

So the fix he suggested for ShutdownThread I've also applied to the QueuedThreadPool, and also rolled the fix for the ShutdownThread into this PR instead of opening a separate PR for it. Please (re)review.

**Review by sbordet — APPROVED** [review]

**Review by sbordet — APPROVED** [review]

**Pull request merged**